### ISSUE TITLE
Training Failed

### ISSUE THREAD

**Issue by leroyjware** [mingfang/docker-predictionio#16]
When trying to run the quickstart, I'm getting the error below (found in pio.log). Any idea why the DNS name is not found?

2015-11-14 23:18:39,676 WARN  org.apache.hadoop.hbase.mapreduce.TableInputFormatBase [main] - Cannot resolve the host name for c5681028a703/172.17.0.2 because of javax.naming.NameNotFoundException: DNS name not found [response code 3]; remaining name '2.0.17.172.in-addr.arpa'
2015-11-14 23:18:55,083 ERROR io.prediction.tools.console.Console$ [main] - No valid engine instance found for engine wZ9ZtKGsSc2RwKtbEvxRMUu8BegGYQ4M 5563d8a47b04a95e6c5eb21e9234dda9d8166bed.
Try running 'train' before 'deploy'. Aborting.

**Comment by LouP2** [comment]
I have the same or similar issue, it seems I am read to go but then get the below error message when I try to deploy the engine

root@98766c86d1e7:/quickstartapp# cat pio.log
2015-12-13 22:37:31,587 INFO  io.prediction.tools.console.Console$ [main] - Inspecting PredictionIO...
2015-12-13 22:37:31,596 INFO  io.prediction.tools.console.Console$ [main] - PredictionIO 0.9.4 is installed at /PredictionIO
2015-12-13 22:37:31,599 INFO  io.prediction.tools.console.Console$ [main] - Inspecting Apache Spark...
2015-12-13 22:37:31,623 INFO  io.prediction.tools.console.Console$ [main] - Apache Spark is installed at /spark
2015-12-13 22:37:31,671 INFO  io.prediction.tools.console.Console$ [main] - Apache Spark 1.4.1 detected (meets minimum requirement of 1.3.0)
2015-12-13 22:37:31,673 INFO  io.prediction.tools.console.Console$ [main] - Inspecting storage backend connections...
2015-12-13 22:37:31,712 INFO  io.prediction.data.storage.Storage$ [main] - Verifying Meta Data Backend (Source: ELASTICSEARCH)...
2015-12-13 22:37:34,340 INFO  io.prediction.data.storage.Storage$ [main] - Verifying Model Data Backend (Source: LOCALFS)...
2015-12-13 22:37:34,356 INFO  io.prediction.data.storage.Storage$ [main] - Verifying Event Data Backend (Source: HBASE)...
2015-12-13 22:37:36,232 INFO  io.prediction.data.storage.Storage$ [main] - Test writing to Event Store (App Id 0)...
2015-12-13 22:37:36,308 INFO  io.prediction.data.storage.hbase.HBLEvents [main] - The namespace pio_event doesn't exist yet. Creating now...
2015-12-13 22:37:36,839 INFO  io.prediction.data.storage.hbase.HBLEvents [main] - The table pio_event:events_0 doesn't exist yet. Creating now...
2015-12-13 22:37:37,892 INFO  io.prediction.data.storage.hbase.HBLEvents [main] - Removing table pio_event:events_0...
2015-12-13 22:37:39,600 INFO  io.prediction.tools.console.Console$ [main] - (sleeping 5 seconds for all messages to show up...)
2015-12-13 22:37:44,609 INFO  io.prediction.tools.console.Console$ [main] - Your system is all ready to go.
root@98766c86d1e7:/quickstartapp# cd MyRecommendation && pio deploy --ip 0.0.0.0&
[2] 6867
root@98766c86d1e7:/quickstartapp# [ERROR] [Console$] No valid engine instance found for engine iEPE3vjlJt0fltrcGptTNcYYy7Bx2C3H d226d35564d904d3c6610458191c461c1b2fc9ff.
Try running 'train' before 'deploy'. Aborting.

[2]+  Exit 1                  cd MyRecommendation && pio deploy --ip 0.0.0.0

**Comment by cwestin** [comment]
I'm also having this problem, in a virtualbox VM I'm using to experiment with. I found this other ticket with the same problem: https://github.com/mingfang/docker-predictionio/issues/10 . I've tried the suggested solution with dnsmasq, but it doesn't seem to help. Still can't deploy. Is there a way to find out the deployed engine instance ids? pio app list shows the access keys, and the string names from engine.json, but it's not clear how that relates to the engine id and version that come out with this error message "No valid engine instance found for engine iEPE3vjlJt0fltrcGptTNcYYy7Bx2C3H d226d35564d904d3c6610458191c461c1b2fc9ff" In my case, those values match the ones in manifest.json, but I don't know where those came from.
